ext/sdk/resources/sdk-root/shell/src/resource-templates/ts/scaffolder.ts

```diff
@@ -0,0 +1,171 @@
+        )));
+
+    // Server Script
+    promises.push(this.fsService.mkdirp(relativePath('server'))
+      .then(() => concurrently(
+          this.fsService.writeFile(relativePath('server/server.ts'), `console.log("[${resourceName}] Server Resource Started");`),
+          this.fsService.writeFile(relativePath('server/tsconfig.json'), getServerTSConfig()),
+        )));
+
+    // Build Tools
+    promises.push(this.fsService.writeFile(relativePath('package.json'), getPackageJson(resourceName)));
+    promises.push(this.fsService.writeFile(relativePath('build.js'), getBuildJS()));
+    promises.push(this.fsService.writeFile(relativePath('.fxdkignore'), getIgnore()));
+
+    await Promise.all(promises);
+
+    await this.installModules(resourcePath);
+  }
+
+  private async installModules(path: string) {
+    return new Promise<void>((resolve, reject) => {
+      cp.exec("yarn", { cwd: path, windowsHide: true }, (err) => {
+        if (err) {
+          return reject(err);
+        }
+
+        return resolve();
+      });
+    });
+  }
+}
+
+function getIgnore(): string {
+  return `client/
+server/
+node_modules/
+package.json
+build.js
+`;
+}
+
+function getClientTSConfig(): string {
+  return `{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "*": ["types/*"]
+    },
+    "outDir": "./",
+    "noImplicitAny": true,
+    "module": "commonjs",
+    "target": "es2016",
+    "lib": ["es2017"],
+    "types": ["@citizenfx/client", "@types/node"],
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "esModuleInterop": true,
+    "noEmit": true,
+  },
+  "include": ["./**/*", "../common/**/*"],
+  "exclude": ["**/node_modules"]
+}`;
+}
+
+function getServerTSConfig(): string {
+  return `{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "*": ["types/*"]
+    },
+    "outDir": "./",
+    "noImplicitAny": true,
+    "module": "commonjs",
+    "target": "es2016",
+    "allowJs": false,
+    "lib": ["es2015", "DOM"],
+    "types": ["@citizenfx/server", "@types/node"],
+    "moduleResolution": "node",
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "resolveJsonModule": true,
+    "esModuleInterop": true,
+    "noEmit": true
+  },
+  "include": ["./**/*", "../common/**/*"],
+  "exclude": ["**/node_modules"]
+}`;
+}
+
+function getPackageJson(resourceName: string): string {
+  return `{
+  "name": "${resourceName}",
+  "version": "1.0.0",
+  "main": "index.js",
+  "author": "you",
+  "license": "MIT",
+  "scripts": {
+    "build": "cross-env MODE=production node build.js",
+    "watch": "cross-env MODE=development node build.js"
+  },
+  "dependencies": {},
+  "devDependencies": {
+    "@citizenfx/client": "latest",
+    "@citizenfx/server": "latest",
+    "@types/node": "^16.3.1",
+    "cross-env": "7.0.3",
+    "esbuild": "^0.14.10"
+  }
+}`;
+}
+
+function getBuildJS(): string {
```

Comment:
Are you sure this will work properly for both client/server? I would've assumed we needed to define a platform for the server-side scripts? And can be done only with the CLI perhaps? 

Like `esbuild serverr/server/js --bundle --platform=node --target=someTarget --outfile=dist/server.js`

We also might not want to minify server-side, or what do you think?

Reply:
Realistically it would be up to the end user if they wanna minify their server, I could see the justification for not minifying it by default.

As far as defining the platform as node, I've never had issues using this setup. Are there specific use cases that you think won't be covered?

Reply:
While it is much more appreciated to not include "junk" build files in resource itself, there's only two options to achieve parallel build of both server and client parts: extra dep like `concurrently` or programmatically with such file, I prefer the latter.